src/main/java/org/dependencytrack/tasks/scanners/SnykAnalysisTask.java

```diff
@@ -208,9 +220,9 @@ public void analyze(final List<Component> components) {
                         countDownLatch.countDown();
                     }
                 };
+                analysisUtil = RateLimiter.decorateRunnable(limiter, analysisUtil);
```

Comment:
This will also apply the rate limit for cached results, because the cache lookup is happening inside `analysisUtil`.

Reply:
moved the cache lookup outside the analysisutil